### DISCUSSION TITLE
Backend routes starts with api is served by next router

### DISCUSSION THREAD

**Discussion by rajatdhoot123** [vercel/next.js#13436] [Help]
# Bug report
Client-side routes are served by the custom node server
## Describe the bug

Here is my custom server code
```
const express = require("express");
const next = require("next");

const port = parseInt(process.env.PORT, 10) || 8080;
const env = process.env.NODE_ENV;
const dev = env !== "production";
const app = next({ dev });

const handle = app.getRequestHandler();

let server;
app
  .prepare()
  .then(() => {
    server = express();

    server.all("*", (req, res) => handle(req, res));

    server.listen(port, err => {
      if (err) {
        throw err;
      }
      console.log(`> Ready on port ${port} [${env}]`);
    });
  })
  .catch(err => {
    console.log("An error occurred, unable to start the server");
    console.log(err);
  });
```
My backend routes start from /api/** 

When I am accessing the server-side rendered page it was working properly.
But when I am making routing on client side it was broken don't know what is wrong here.
One Observation even when I am trying to hit backend routes through the browser it was assuming it as next js app and there was no route match it was returning 404.

## Expected behavior

It should not consider backend routes as next js app

## System information

- OS: [e.g. macOS, Windows]
- Browser (if applies) [e.g. chrome, safari]
- Version of Next.js: 9.4.2
- Version of Node.js: v10.19.0

**Comment by tcelestino** [comment]
There's a discussion about it. https://github.com/vercel/next.js/pull/9872
